### PR TITLE
feat(skip): Allow to skip formatter, linter or unit tests by setting Env variables

### PR DESCRIPTION
**Pull request by benoitf**
### What does this PR do?
allow to skip formatter, linter or tests by specifying env variables
`SKIP_FORMAT` `SKIP_TEST` `SKIP_LINT` to true

### What issues does this PR fix or reference?
https://github.com/eclipse/che/issues/18451

### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)

- [x] [The Eclipse Contributor Agreement is valid](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-eclipse-contributor-agreement-is-valid)
- [x] [Code produced is complete](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-produced-is-complete)
- [x] [Code builds without errors](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-builds-without-errors)
- [x] [Tests are covering the bugfix](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#tests-are-covering-the-bugfix)
- [x] [The repository devfile is up to date and works](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-repository-devfile-is-up-to-date-and-works)
- [x] [Sections `What issues does this PR fix or reference` and `How to test this PR` completed](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#sections-what-issues-does-this-pr-fix-or-reference-and-how-to-test-this-pr-completed)
- [ ] [Relevant user documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [Relevant contributing documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [x] [CI/CD changes implemented, documented and communicated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#cicd-changes-implemented-documented-and-communicated)

### Reviewers

Reviewers, please comment how you tested the PR when approving it.

### Happy Path Channel

HAPPY_PATH_CHANNEL=stable

Change-Id: Iba5968f6883a50e17aa63a65155a9b2450507e53
Signed-off-by: Florent Benoit <fbenoit@redhat.com>

### CODE REVIEW

File: dockerfiles/theia/Dockerfile [hunk 1]
```diff
@@ -27,6 +27,9 @@ ARG THEIA_COMMIT_SHA=''
 
 ENV NODE_OPTIONS="--max-old-space-size=4096"
 
+# avoid any linter/formater/unit test
+ENV SKIP_LINT=true SKIP_FORMAT=true SKIP_TEST=true
```
Comment (tsmaeder) [inline dockerfiles/theia/Dockerfile:31]: I believe the default should be `false`. Why would we build docker images without checking, but not allow local build without it?

**Review by tsmaeder — COMMENTED** [review]
IMO, the default should be "no-check" on local builds but a PR check for correct linting and formatting: the usual development workflow is 
1. Make changes for for a week
2. Run the formatter once
3. Make a PR.

**Comment by tsmaeder** [comment]
IMO, the linter warnings and formatting problems should not prevent the build from proceeding. We can have a PR check that disallows the PR from being merged if formatting and linter warnings exist.

File: dockerfiles/theia/Dockerfile (hunk 1 above)
Comment (benoitf) [reply dockerfiles/theia/Dockerfile:31]: because it's checked on another job (and it allows to make the builds as fast as possible)

**Comment by benoitf** [comment]
Here the default is 'check/lint/test' by default as it's done by maven (it runs tests by default unless you say you want to skip, etc) and in che project in general. (-Pfast)

But we could add skip options in theia-dev images to have dogfooding with default=skip

should be a vote on what people prefer by default:
- checks turn on by default
- checks turn off by default

**Comment by RomanNikitenko** [comment]
> should be a vote on what people prefer by default:
>- checks turn on by default
>- checks turn off by default

I think such checks should be turned ON by default

**Review by azatsarynnyy — APPROVED** [review]
I believe all checks have to be turned on, by default.
Anyway, approving the PR as it's configurable.

**Review by RomanNikitenko — APPROVED** [review]

**Comment by ericwill** [comment]
IMO there should be a PR check to do all checking once a PR is pushed. What people want to do on their own local machine is up to them. So my vote is to have checks off by default, and create a PR check that enables them and does the build. People who like the extra strictness are always free to enable checking on their local machine.

**Comment by benoitf** [comment]
Just a FYI, I don't know if it may help on voting.
The upstream project, eclipse theia has linter check turn on by default.
so even if we disable checks on che-theia, it will still do the checks on modules of eclipse-theia (I want to propose the same way to allow to disable checks) but then it may result in having odd behaviour if eclipse-theia and che-theia don't have the same defaults

**Review by vitaliy-guliy — APPROVED** [review]

**Comment by benoitf** [comment]
Upstream theia has linter being enabled by default (I've added in upstream Theia SKIP_LINT parameter as well), so by following the same convention in che-theia it's easier to have linter being enabled by default and have skip options.

Merging

**Pull request merged**